packages/foam-vscode/src/extension.ts

```diff
@@ -30,7 +30,7 @@ export async function activate(context: ExtensionContext) {
     const foamPromise: Promise<Foam> = bootstrap(config, services);
 
     features.forEach(f => {
-      f.activate(context, foamPromise);
+      f.activate(context, foamPromise, dataStore);
```

Comment:
to pass the data store along, I would suggest either using the `context`, or passing the whole `services` object, so that in the future we don't need to keep adding fields to this call - I'd suggest the former (after all that's what the context is meant for), what do you think?

Reply:
By using the context, how would you pass it? By finding the data store in `context.subscriptions`? Because if not we only have a global and a workspace data store. So maybe passing the `services` is best?

Reply:
what I meant is that we can have a `FoamContext` which extends the `ExtensionContext` that we are getting.

```
c: FoamContext = {
...context,
dataStore: ..
}
```

and that's what we pass to the various feature activations, does that make sense?

Reply:
makes sense! both of your comments should be addressed now!